### PR TITLE
drivers: sensor: grow_r502a: Add Missing functionalities that R502a provides

### PR DESCRIPTION
**Pull request by DineshDK03**
Add the following functionality of the R502A fingerprint sensor device.

1. Add upload and download of fingerprint template data.
2. Add LED control for the R502A sensor's built-in LED.
3. Add Read and write system parameters of the R502A sensor.
4. Add precise matching of the given index with a provided finger.
5. Add capture, load, store, and compare templates of fingers with the R502A sensor.

### CODE REVIEW

File: drivers/sensor/grow_r502a/Kconfig [hunk 1]
```diff
@@ -13,6 +13,12 @@ menuconfig GROW_R502A
 
 if GROW_R502A
 
+config GROW_R502A_DATA_PKT_SIZE
+	int "Template data packet size"
+	default 128
```
Comment (parthitce) [inline drivers/sensor/grow_r502a/Kconfig:18]: Do we need this configurable? Why?

Comment (parthitce) [inline drivers/sensor/grow_r502a/Kconfig:18]: Do we need this configurable? Why?

**Comment by parthitce** [comment]
Testing the webhook

**Comment by parthitce** [comment]
Testing webhook 1

**Comment by parthitce** [comment]
Test webhook

**Comment by parthitce** [comment]
Testing 1

**Comment by parthitce** [comment]
Test

File: drivers/sensor/grow_r502a/grow_r502a.h [hunk 2]
```diff
@@ -182,7 +171,12 @@ struct grow_r502a_data {
 #endif /* CONFIG_GROW_R502A_TRIGGER */
 
 	struct r502a_buf tx_buf;
-	struct r502a_buf rx_buf;
+
+	struct r502a_buf rx_data_buf;
+	union r502a_packet rx_hdr_pkt;
+	int rx_len;
+	bool rx_header;
```
Comment (ssekar15) [inline drivers/sensor/grow_r502a/grow_r502a.h:178]: i repeat remove rx-header... first do self review and remove unnecessary code

Comment (DineshDK03) [reply drivers/sensor/grow_r502a/grow_r502a.h:178]: Okay will do that, thanks for the review.

**Comment by DineshDK03** [comment]
closing this for now as the same requirement for this driver was developed with new logic and opened as a new PR here https://github.com/linumiz/zephyr/pull/12, can be reopened later if needed.

**Pull request closed without merging**